### PR TITLE
fix(parallelCollectionScan): do not use implicit sessions on cursors

### PR DESCRIPTION
**Pull request by daprahamian**
When we removed the session use from parallelCollectionScan,
we introduced an error where parallelCollectionScan cursors would
attempt to use implicit sessions on their cursors.

Looking for feedback on the name of the field :)

### CODE REVIEW

File: lib/cursor.js [hunk 1]
```diff
@@ -143,8 +143,10 @@ function Cursor(bson, ns, cmd, options, topology, topologyOptions) {
     promiseLibrary: promiseLibrary,
     // Current doc
     currentDoc: null,
+    // expicitlyDoNotUseSession
+    expicitlyDoNotUseSession: options.expicitlyDoNotUseSession,
```
Comment (kvwalker) [inline lib/cursor.js]: typo: `expicitly` should be `explicitly`

File: lib/cursor.js [hunk 2]
```diff
@@ -211,7 +213,7 @@ for (let name in CoreCursor.prototype) {
 }
 
 Cursor.prototype._initImplicitSession = function() {
-  if (!this.s.session && this.s.topology.hasSessionSupport()) {
+  if (!this.s.expicitlyDoNotUseSession && !this.s.session && this.s.topology.hasSessionSupport()) {
```
Comment (kvwalker) [inline lib/cursor.js]: same typo

File: lib/operations/collection_ops.js [hunk 3]
```diff
@@ -1021,6 +1021,8 @@ function parallelCollectionScan(coll, options, callback) {
         null
       );
 
+    options = Object.assign({ expicitlyDoNotUseSession: true }, options);
```
Comment (kvwalker) [inline lib/operations/collection_ops.js]: same typo

**Comment by mbroadst** [comment]
Can you please include a test for this?

File: lib/cursor.js (hunk 1 above)
Comment (mbroadst) [inline lib/cursor.js]: we already have a `skipSessions` elsewhere in the code, would it make sense to keep naming consistent even though they are in disparate locations?

**Comment by daprahamian** [comment]
@mbroadst I can verify this works b/c there were ~8 existing tests that were failing.

Do you want a test to explicitly check that we are not using implicit sessions?

File: lib/cursor.js (hunk 1 above)
Comment (daprahamian) [reply lib/cursor.js]: I kind of like @kvwalker's `explicitlyIgnoreSession`. I want to make sure we do not confuse the `skipSession` variable with this.

**Comment by mbroadst** [comment]
@daprahamian the way I see it there are two things happening here: you are adding a feature to cursors that allow us explicitly ignore sessions, and then modifying the `parallelCollectionScan` to use that feature. Are we potentially going to use that feature for other methods in the future? Maybe. If so, I think it's better to validate that this thing does what we expect it to from the cursor level, rather than depend on side-effecty breaks in other tests that might subtly change in the future and no longer provide us the same coverage. Unless, of course, one of those eight tests _is_ explicitly testing this, then I retract my statement 😄

File: lib/cursor.js (hunk 1 above)
Comment (mbroadst) [reply lib/cursor.js]: SGTM

**Pull request merged**